diff -uNr rtl8180-0.15.orig/ieee80211_module.c rtl8180-0.15/ieee80211_module.c
--- rtl8180-0.15.orig/ieee80211_module.c	2004-11-16 01:17:45.000000000 +0100
+++ rtl8180-0.15/ieee80211_module.c	2004-12-25 12:25:07.094112240 +0100
@@ -227,7 +227,7 @@
 	INIT_LIST_HEAD(&ieee->beacons);
 	
 	init_waitqueue_head(&ieee->assoc_queue);
-	ieee->workqueue = create_workqueue(DRV_NAME);
+	ieee->workqueue = create_workqueue(DRV_NAME,PF_SYNCTHREAD);
 	INIT_WORK(&ieee->associate_tasklet,(void(*)(void*)) ieee80211_associate_tasklet,ieee);
 	
 
diff -uNr rtl8180-0.15.orig/Makefile rtl8180-0.15/Makefile
--- rtl8180-0.15.orig/Makefile	2004-11-13 20:44:11.000000000 +0100
+++ rtl8180-0.15/Makefile	2004-12-25 12:55:23.973904528 +0100
@@ -4,7 +4,7 @@
 PWD = $(shell pwd)
 KVER := $(shell uname -r)
 KSRC := /lib/modules/$(KVER)/build
-include $(KSRC)/.config
+include .config
 INSTALL_PREFIX :=
 EXTRA_CFLAGS += -std=gnu89
 EXTRA_CFLAGS += -O2
